### PR TITLE
SvelteKit release candidate migration: events

### PR DESCRIPTION
**Pull request by EthanThatOneKid**
Updated package.json
- Ran `npm i @js-temporal/polyfill@^0.4.1`
- Ran `npm i rrule@^2.6.9` (declared module in src/app.d.ts)

Migrated...
- New icons to /src/lib/components/svg/
- Page endpoint to src/routes/events/+page.svelte, src/routes/events/+page.ts
- JSON endpoint to src/routes/events/[query].json/+server.ts
- Sample data to src/lib/server/events/data/
- In-memory cache in src/lib/server/events/cache.ts
  - Created common cache service in src/lib/server/cache/cache.ts
- Event list component in src/routes/events/list.svelte
  - Event item component in src/routes/events/event.svelte
- iCal logic to src/lib/server/events/ical.ts
- iCal test to src/lib/server/events/ical.test.ts

### CODE REVIEW

File: src/lib/server/events/cache.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import type { ClubEvent } from '$lib/public/events/event';
+import { Cache } from '$lib/server/cache/cache';
+
+// Make this false to disable server-side caching in development.
+export const allEvents = Cache.create<ClubEvent[]>(/* CACHING=*/ true);
```
Comment (EthanThatOneKid) [inline src/lib/server/events/cache.ts:5]: In the future, we may want to cache the events data on the server when "debug mode" is DISABLED. This allows the user to only have to change the value of DEBUG_FLAG_ENABLED in their `.env`.
```suggestion
import { DEBUG_FLAG_ENABLED } from '$lib/server/flags';

export const allEvents = Cache.create<ClubEvent[]>(/* CACHING=*/ !DEBUG_FLAG_ENABLED);
```

**Review by jaasonw — COMMENTED** [review]
image
line break issue on long location when window small ish, should i make a new issue for this or request changes here?

**Comment by EthanThatOneKid** [comment]
> line break issue on long location when window small ish, should i make a new issue for this or request changes here?

A separate acmcsuf.com/bug would be appropriate since this PR is primarily for migrating the current events page features.

**Review by jaasonw — APPROVED** [review]
ok

**Comment by EthanThatOneKid** [comment]
Replaced by mega-PR #577.

**Pull request closed without merging**